crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -224,28 +226,43 @@ where
         &mut self,
         input: &Message,
     ) -> Result<Vec<Message>, ConversionError> {
-        let tedge_event = ThinEdgeEvent::try_from(input.topic.name.as_str(), input.payload_str()?)?;
-        let c8y_event = C8yCreateEvent::try_from(tedge_event)?;
+        let mut vec = Vec::new();
+        let c8y_event;
+
+        let tedge_event = ThinEdgeEvent::try_from(&input.topic.name, input.payload_str()?)?;
+        let child_id = tedge_event.source.clone();
+
+        Self::create_external_source_if_does_not_exist(self, &tedge_event, &mut vec);
+
+        // Convert the external source event message to Cumulocity JSON message.
+        c8y_event = C8yCreateEvent::try_from(tedge_event)?;
 
         // If the message doesn't contain any fields other than `text` and `time`, convert to SmartREST
         let message = if c8y_event.extras.is_empty() {
             let smartrest_event = Self::serialize_to_smartrest(&c8y_event)?;
             let smartrest_topic = Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC);
-
             Message::new(&smartrest_topic, smartrest_event)
         } else {
             // If the message contains extra fields other than `text` and `time`, convert to Cumulocity JSON
             let cumulocity_event_json = serde_json::to_string(&c8y_event)?;
             let json_mqtt_topic = Topic::new_unchecked(C8Y_JSON_MQTT_EVENTS_TOPIC);
-
             Message::new(&json_mqtt_topic, cumulocity_event_json)
         };
 
         // If the MQTT message size is well within the Cumulocity MQTT size limit, use MQTT to send the mapped event as well
         if input.payload_bytes().len() < self.size_threshold.0 {
-            Ok(vec![message])
+            vec.push(message);
+            Ok(vec)
+        // If the message size is larger than the MQTT size limit, use HTTP to send the mapped event
+        } else if let Some(id) = child_id {
+            if self.children.contains(&id) {
+                let _ = self.http_proxy.send_event(c8y_event).await?;
+                Ok(vec![])
+            } else {
+                Err(ConversionError::ChildDeviceNotRegistered { id: id.to_string() })
+            }
         } else {
-            // If the message size is larger than the MQTT size limit, use HTTP to send the mapped event
+            // Parent device
             let _ = self.http_proxy.send_event(c8y_event).await?;
             Ok(vec![])
         }
```

Comment:
The code to send the event over HTTP doesn't depend on the child id - and must be outside the loop.

Furthermore this code is buggy. Since the child id has been added by `create_external_source_if_does_not_exist`, only the `else` case is used.